features/admin.server-configurations.v1/forms/password-recovery-form/password-recovery-form.tsx

```diff
@@ -0,0 +1,802 @@
+ * @returns Functional component.
+ */
+export const PasswordRecoveryConfigurationForm: FunctionComponent<PasswordRecoveryConfigurationFormPropsInterface> = (
+    props: PasswordRecoveryConfigurationFormPropsInterface
+): ReactElement => {
+
+    const {
+        initialValues,
+        onSubmit,
+        readOnly,
+        isConnectorEnabled,
+        isSubmitting,
+        ["data-testid"]: testId
+    } = props;
+
+    const { t } = useTranslation();
+    const [ initialConnectorValues, setInitialConnectorValues ]
+        = useState<PasswordRecoveryFormInitialValuesInterface>(undefined);
+    const [ isEmailRecoveryEnabled, setIsEmailRecoveryEnabled ] = useState<boolean>(false);
+    const [ isSMSRecoveryEnabled, setIsSMSRecoveryEnabled ] = useState<boolean>(false);
+    const [ isUpperCaseEnabled, setIsUpperCaseEnabled ] = useState<boolean>(false);
+    const [ isLowerCaseEnabled, setIsLowerCaseEnabled ] = useState<boolean>(false);
+    const [ isNumericEnabled, setIsNumericEnabled ] = useState<boolean>(false);
+    const showSmsOtpPasswordRecoveryBetaChip: boolean =
+        useSelector((state: AppState) => state?.config?.ui?.showSmsOtpPasswordRecoveryBetaChip);
+
+    /**
+     * Flattens and resolved form initial values and field metadata.
+     */
+    useEffect(() => {
+
+        if (isEmpty(initialValues?.properties)) {
+            return;
+        }
+
+        let resolvedInitialValues: PasswordRecoveryFormInitialValuesInterface = null;
+
+        initialValues.properties.map((property: ConnectorPropertyInterface) => {
+            if (allowedConnectorFields.includes(property.name)) {
+                if (property.name === ServerConfigurationsConstants.NOTIFY_SUCCESS) {
+                    resolvedInitialValues = {
+                        ...resolvedInitialValues,
+                        notifySuccess: CommonUtils.parseBoolean(property.value)
+                    };
+                } else if (property.name === ServerConfigurationsConstants.RECOVERY_LINK_EXPIRY_TIME) {
+                    resolvedInitialValues = {
+                        ...resolvedInitialValues,
+                        expiryTime: property.value
+                    };
+                } else if (property.name === ServerConfigurationsConstants.RECOVERY_EMAIL_LINK_ENABLE) {
+                    resolvedInitialValues = {
+                        ...resolvedInitialValues,
+                        enableEmailBasedRecovery: CommonUtils.parseBoolean(property.value)
+                    };
+                } else if (property.name === ServerConfigurationsConstants.RECOVERY_SMS_OTP_ENABLE) {
+                    resolvedInitialValues = {
+                        ...resolvedInitialValues,
+                        enableSMSBasedRecovery: CommonUtils.parseBoolean(property.value)
+                    };
+                } else if (property.name === ServerConfigurationsConstants.RECOVERY_SMS_EXPIRY_TIME) {
+                    resolvedInitialValues = {
```

Comment:
Make this a switch statement for readability

Reply:
These are existing implementations which were not changed through this effort. The changes detect them as new due to the file being moved (to add scss). Do we need to improve these previous implementations through this effort?